packages/block-editor/src/hooks/color.js

```diff
@@ -454,15 +454,20 @@ export const withColorPaletteStyles = createHigherOrderComponent(
 			return <BlockListBlock { ...props } />;
 		}
 
-		const extraStyles = {
-			color: textColor
-				? getColorObjectByAttributeValues( colors, textColor )?.color
-				: undefined,
-			backgroundColor: backgroundColor
-				? getColorObjectByAttributeValues( colors, backgroundColor )
-						?.color
-				: undefined,
-		};
+		const extraStyles = {};
+
+		if ( textColor ) {
+			extraStyles.color = getColorObjectByAttributeValues(
```

Comment:
This is a good fix.

However, inspecting `getColorObjectByAttributeValues` there's a path in which the resulting `color` value still can be `undefined`: when passing a `textColor` that is not present in the list of `colors`. This can happen upon switching themes, for example. Specific steps:

- Use TwentyTwentyTwo.
- Add a paragraph and select a preset text color from the theme.
- Switch to EmptyTheme.

At this point, the text color is undefined. I wonder if `getColorObjectByAttributeValues` should return `null` instead of `undefined` for such a case.

Though at the moment this is not reproducible because we have a bug in `trunk` by which the preset colors are serialized as inline styles and not as classes. Going to follow up https://github.com/WordPress/gutenberg/issues/36770